source/bao_hyp/config.rst

```diff
@@ -3,7 +3,78 @@ Bao Configuration File
+in the form of a C source file. This section provides an in-depth
+description of the various configuration options available.
+
+The configuration file follows a specific structure, as outlined below:
+
+.. code-block:: c
+
+    #include <config.h>
+
+    // Load guests' image
+    VM_IMAGE(img1_name, "/path/to/vm1/binary.bin");
+    VM_IMAGE(img2_name, "/path/to/vm2/binary.bin");
+
+    struct config config = {
+
+        // Shared memories configuration
+        .shmemlist_size = N,
+        .shmemlist = (struct shmem[]) {
+            [0] = {/*shared memory config*/,},
+            [1] = {/*shared memory config*/,},
+            ...
+            [N] = {/*shared memory config*/,}
+        },
+
+        // Guests Configuration
+        .vmlist_size = NUM_VMs,
+        .vmlist = {
+            { /* VM 0 Config*/},
+            { /* VM 1 Config*/},
+            ...
+            { /* VM N Config*/},
+        }
+    };
+
+Before defining the configuration struct, it's crucial to declare the VM
+images. The configuration file anticipates receiving a compiled image of each
+guest (.bin file) which will then be allocated to individual VMs using the
+``VM_IMAGE`` macro. This macro allows for the direct embedding of the guest
+binary file into the hypervisor binary file compiled program. Here's an example
+usage of the ``VM_IMAGE``:
+
+.. code-block:: c
+
+  VM_IMAGE(img_name, "/path/to/VM/binary.bin");
+
+In this context, ``img_name`` is the identifier associated with the image that
+will later be used to describe the image running on the VM - `Guest Image`_.
+
+After loading the images, the configuration allows the definition of two
+distinct lists: (i) a list of shared memory regions (``shmemlist``) and (ii) a
+list of VMs (or guests) (``vmlist``). While the list of shared memory regions
+is optional and may be omitted from the configuration, the list of VMs is
+mandatory and must include at least 1 VM. Additionally, for each list, it is
+necessary to specify the list size using the parameters ``shmemlist_size`` and
+``vmlist_size``.
+
+Config file name and location
+-----------------------------
+The configuration file with the ``.c`` extension can be assigned various names.
+As part of the Bao hypervisor's build system, it expects to find the
+configuration file in the ``configs`` folder. This means that users need to
+place their custom configuration files with the appropriate settings in the
+designated ``configs`` directory. The build system then relies on this file to
+extract and apply the desired configuration options for the hypervisor.
+
+.. warning::
+    Modifying the number of elements in the ``shmemlist`` or \
+    ``vmlist`` lists, which differs from the specified sizes \
+    (``shmemlist_size`` and ``vmlist_size``, respectively), may result in \
+    unpredictable behavior. Ensure that any changes made to the configuration \
+    are consistent with the designated sizes to avoid unexpected issues.
```

Comment:
I would swapo the order of ideas:


```suggestion
    Inconsistences between the specified list sizes \
    (``shmemlist_size`` and ``vmlist_size``, respectively) and the actual sizes of the lists, may result in \
    unpredictable behavior. Ensure that any changes made to the configuration lists' number of elements \
    is reflected in the respective list size.
```

Reply:
Agree, thanks! Addressed in commit #77e63cf